ui

SDeeKey opened the pull request.

SDeeKey committed:
ui

yaa-ku commented:
оно, конечно, и без этого работает, но импортни реакт, чтоб IDE не воняло

![image](https://github.com/user-attachments/assets/6db7d59a-6856-489a-88ad-4e8067210d27)

import React from 'react';

yaa-ku commented on src/app/page.tsx:
```diff
@@ -1,4 +1,3 @@
-import Image from "next/image";
 import styles from "./page.module.css";
 
 export default function Home() {
```
вместо тэга div используй тэг main

у тебя сейчас структура проекта:

```
<html>
    <head/>
    <body>
        <div>...</div> // <- должно быть <main>...</main>
    </body>
</html>
```

![image](https://github.com/user-attachments/assets/e86c9883-1c1e-4cbf-9c8b-088e34b564f4)

[источник почитать](https://developer.mozilla.org/en-US/docs/Learn/HTML/Introduction_to_HTML/Document_and_website_structure#basic_sections_of_a_document)

yaa-ku commented:
меня смущают версии некоторых пакетов:

```
"dependencies": {
    "react": "19.0.0-rc-66855b96-20241106", <- ето что за версия такая страшная
    "react-dom": "19.0.0-rc-66855b96-20241106", <- и ето
    "next": "15.0.3"
  },
  "devDependencies": {
    "typescript": "^5", <- в идеале вообще ставить всегда конкретные версии.
    "@types/node": "^20",
    "@types/react": "^18",
    "@types/react-dom": "^18",
    "eslint": "^8",
    "eslint-config-next": "15.0.3"
  }
```

основной поинт: в идеале в больших проектах лучше фиксировать версию. потому что иначе может установиться сломанный патч или минорная версия. здесь не критично, но лучше держи в голове идею того, что лучше фиксировать

~ и ^ в целом допускаются

yaa-ku commented on src/shared/ui-kit/button/button.module.scss:
```diff
@@ -0,0 +1,19 @@
+}
+.btn:active {
+  background-color: var(--orange500);
+}
```
почитай про вложенные селекторы в sass ([nested selectors](https://sass-lang.com/documentation/style-rules/declarations/#nesting))

```
.btn {
  display: flex;
  background-color: var(--orange400);
  border-radius: 20px;
  color: var(--gray50);
  padding: 8px 48px;
  border: none;
  cursor: pointer;

  &:disabled {
    opacity: 0.4;
  }

  &:hover {
    background-color: var(--orange300);
  }

  &:active {
    background-color: var(--orange500);
  }
}
```

можешь потренироваться выносить в [миксины](https://sass-lang.com/documentation/at-rules/mixin/):
например можно для проекта сделать миксины для флексов:

```
@mixin flex-column($gap: 0){
  display: flex;
  flex-direction: column;
  gap: $gap;
}

@mixin flex-row($gap: 0){
  display: flex;
  flex-direction: row;
  gap: $gap;
}
```

p.s. это просто пример, ты можешь придумать что-то свое, если вдруг тебе придет в голову более удобная реализация

миксины - по факту, функции из кусков переиспользуемого scss кода. можно сделать свою маленькую "библиотеку" миксинов для проекта

например, у меня на проекте они лежат в файле:
src/styles/mixins.scss

не забывай их импортировать

yaa-ku commented on src/shared/ui-kit/button/button.tsx:
```diff
@@ -0,0 +1,14 @@
+
+const Button = ({title}: {title: string}) => {
+    return (
+        <div>
```
лишняя обертка

yaa-ku commented on src/shared/ui-kit/button/button.tsx:
```diff
@@ -0,0 +1,14 @@
+    );
+};
+
+export default Button;
```
можно без export default

а ещё можно пропсы вот так прописать, это более удобно (тебе, к тому же, надо добавить обработчик нажатия на кнопку):
```

interface ButtonProps {
    title: string;
}

export const Button = ({ title }: ButtonProps) => {
    return (
        <button className={styles.btn}>
            <sub className="subtitleSB">{title}</sub>
        </button>
    );
};
```

yaa-ku commented on src/shared/ui-kit/button/button.tsx:
```diff
@@ -0,0 +1,14 @@
+const Button = ({title}: {title: string}) => {
+    return (
+        <div>
+             <button className={styles.btn}>
```
кста, мы модули чутка не так используем обычно

```
import s from './button.module.scss'; <- вкусовщина, просто пример
...
 <button className={s['btn']}> <- тоже пожалуй вкусовщина. но так будет удобно, если у тебя какой-то сложный динамический селектор. и чтоб везде "одинаково" было
```

yaa-ku commented on src/shared/ui-kit/button/button.tsx:
```diff
@@ -0,0 +1,14 @@
+    return (
+        <div>
+             <button className={styles.btn}>
+                <sub className='subtitleSB'>{title}</sub>
```
по семантике нелогично этот селектор использовать

![image](https://github.com/user-attachments/assets/70688306-3ef4-4b95-a62d-caa42afabb41)

подробнее почитай [тут](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/sub)

yaa-ku replied:
вообще можно вложить в p. или в целом сделать просто

```
<button className={*какие-то стили для кнопки*}>
    {label}
</button>
```

yaa-ku commented on src/shared/ui-kit/linkIcon/linkIcon.module.scss:
```diff
@@ -0,0 +1,13 @@
+.linkIconContainer {
+  display: flex;
+  align-items: center;
+  justify-content: center;
+  background-color: var(--orange400);
+  border-radius: 8px;
+}
+.linkIconContainer:hover {
+  background-color: var(--orange300);
+}
+.linkIconContainer:active {
+  background-color: var(--orange500);
+}
```
добавь вложенности селекторам

yaa-ku commented on src/shared/ui-kit/linkIcon/linkIcon.tsx:
```diff
@@ -0,0 +1,30 @@
+}
+
+const LinkIcon: React.FC<LinkIconProps> = ({ link, size }) => {
+    const sizes: Record<string, React.CSSProperties> = { // Определяем тип объекта, который используем
+        md: { height: '36px', width: '36px' },
+        lg: { height: '44px', width: '44px' },
+    };
```
inline стили - гейство. особенно если неоправданное

передавай через пропсы вариант отображения компонента. передавай соответствующий стиль для компонента, чтобы подключался нужный стиль из scss файлика

упрощенный пример:

**button.tsx**

```
interface ButtonProps {
    title: string;
    variant?: 'accent' | 'primary'; <- не обязательно передавать. в случае, если ты не передашь, выставиться значение по умолчанию
}

export const Button = (props: ButtonProps) => {

const {
    title,
    variant = 'primary', <- если параметр будет не передан, то по умолчанию выставиться в 'primary'
} = props; <- ещё один способ, как передать пропсы. тоже удобный

    return (
        <button className={clsx(
           'button', <- стиль для кнопки в целом
           `button--${variant}` <- в зависимости от переданного пропса будет выставляться нужный стиль для кнопки
        )}>
            {title}
        </button>
    );
};
```

**button.module.scss**

```
.button {
// общие стили для кнопки. размер, форма, различные состояния и тд

 &--primary { <- если variant = 'primary' будет красная
     background-color: $red;
     
     &:hover {
         background-color: $dark-red;
     }
 }
 
 &--accent { <- если variant = 'accent' будет зеленая
     background-color: green;
     
     &:hover {
         background-color: $dark-green;
     }
 }
}
```

yaa-ku replied:
я написала пример для цвета, но с размером и прочими настройками отображения - та же ситуация

yaa-ku commented on src/shared/ui-kit/linkIcon/linkIcon.tsx:
```diff
@@ -0,0 +1,30 @@
+    const style = sizes[size] || sizes.md; // По умолчанию размер md
+
+    return (
+        <div>
```
лишняя обертка

yaa-ku commented on src/shared/ui-kit/linkIcon/linkIcon.tsx:
```diff
@@ -0,0 +1,30 @@
+        <div>
+            <a href={link} target='_blank' rel='noopener noreferrer'>
+                <div className={styles.linkIconContainer} style={style}>
+                    <img src="linkIcon.svg" alt="linkIcon" />
```
чет мне не нравится тут img. вроде у Next'а не так работают с картинками/иконками

попробуй поразбираться. я потом напишу комментарий мб

yaa-ku replied:
насчёт [svg + Next.js](https://dev.to/dolearning/importing-svgs-to-next-js-nna)

как я писала в комментах выше, Next содержит в себе особую работу с картинками. чтобы оптимизировать и улучшить опыт их использования в проекте. это делается с помощью компоненте Image

однако несмотря на то, что этот компонент можно подружить с svg, так лучше не делать

**как можно сделать?**
не то чтобы это какое-то ультимативное решение и неиспользование его будет ошибкой, но могу предложить интересный вариант, который мы используем на большом количестве проектов

я могу тебе скинуть рабочую настройку, но как будто лучше чтоб ты сделал это сам, я только наведу на верный путь. но если будут проблемы - напишешь

1. [установи @svgr/webpack](https://www.npmjs.com/package/@svgr/webpack)
2. настрой next.config.js (у тебя вроде next.config.ts. думаю и так можно)
3. импортируй svg и используй как компоненты

пример:
```
import Star from './star.svg'

const App = () => (
  <div>
    <Star />
  </div>
)
```

p.s. как будто лучше хранить svg и прочие картиночки где-нибудь внутри папки App. public больше для шрифтов, фавиконок

yaa-ku commented on src/shared/ui-kit/tag/tag.tsx:
```diff
@@ -0,0 +1,13 @@
+    );
+};
+
+export default Tag;
```
лишний export default Tag

yaa-ku commented:
ты не совсем верно подключил шрифты. они работают, да, все ок, но раз в рамках данного проекта используется Next.js - грешно подключать шрифты, не используя сахара, который превносит Next.js :D  

в дальнейших комментариях я буду описывать, что именно и в каком моменте было сделано не так, а потом объясню, как надо сделать и скину ссылки на документацию с примером

yaa-ku commented on src/app/layout.tsx:
```diff
@@ -25,7 +20,7 @@ export default function RootLayout({
   return (
     <html lang="en">
-      <body className={`${geistSans.variable} ${geistMono.variable}`}>
+      <body className={`${inter.variable}`}>
```
во-первых, вот эта конструкция излишне 

если ты напишешь `<body className={inter.variable}>` все так же будет работать

yaa-ku commented on src/app/layout.tsx:
```diff
@@ -25,7 +20,7 @@ export default function RootLayout({
   return (
     <html lang="en">
-      <body className={`${geistSans.variable} ${geistMono.variable}`}>
+      <body className={`${inter.variable}`}>
```
кста, фан факт: если ты сейчас полностью уберешь из layout.tsx подключение шрифтов, они все равно подключатся :D 

ты их неправильно подключаешь потому что. на самом деле, сейчас твое подключение шрифта держится на строчке 
 ` font-family: "Inter", sans-serif;`
в файле **globals.css**

если закомментишь эту строчку там - все сломается. так не должно быть

yaa-ku replied:
как надо: чтобы починить то, как у тебя получается локальный шрифт - нужно изменить то, что ты устанавливаешь в className в body

`<body className={inter.variable}>` <-❌ не работает, подключается только через globals.css. тоже вариант, но как-то мех, в документации обычно по-другому делают

`<body className={inter.className}>` <- ✔работает и без подключения в globals.css

yaa-ku replied:
подробнее о корректном подключении локального шрифта почитай вот [тут](https://nextjs.org/docs/app/building-your-application/optimizing/fonts#local-fonts) в документации. там есть пример

yaa-ku replied:
то как ты сделал кстати считается правильным для работы с Tailwind, вот [тут](https://nextjs.org/docs/app/building-your-application/optimizing/fonts#with-tailwind-css) подсмотрела. но не для Tailwind так лучше не делать

yaa-ku commented on src/app/layout.tsx:
```diff
@@ -2,19 +2,14 @@ import type { Metadata } from "next";
-  src: "./fonts/GeistMonoVF.woff",
-  variable: "--font-geist-mono",
-  weight: "100 900",
+const inter = localFont({
+  src: "./fonts/Inter.ttf",
+  variable: "--Inter",
+  weight: "400 500 600 700",
 });
```
теперь пойдёт более ✨сложная математика✨ 

ты в целом верно подключил шрифт _локально_, однако согласно документации NextJS лучше использовать другой метод подключения шрифтов. тут в документации как раз есть [пример для этого же шрифта](https://nextjs.org/docs/app/building-your-application/optimizing/fonts#google-fonts)

**зачем?** такой метод описан в документации. это хорошо оптимизировано

> This new font system also allows you to conveniently use all Google Fonts with performance and privacy in mind. CSS and font files are downloaded at build time and self-hosted with the rest of your static assets. No requests are sent to Google by the browser.

**как тогда сделать?** я предлагаю следующий вариант и комментарии к нему:

```
import { Inter } from 'next/font/google'

const inter = Inter({
    subsets: ['latin', 'cyrillic'], <- определить только нужные нам subsets (подробнее в доке)
    display: 'swap', <- определяет то, как будет подключаться шрифт. ❗❗ очень важно почитай про свойство font-display. лучше использовать именно swap
    style: ['normal'], <- тоже можно тут определить
})

// some awesome code

<html lang="en" className={inter.className}> <- сразу применит шрифт ко всему проекту. дополнительно подключать не требуется
```

**почему не указан weight?** 

> If loading a variable font, you don't need to specify the font weight (согласно документации NextJS)

> Inter is a variable font family carefully crafted & designed for computer screens (согласно [инфы о шрифте от Google Fonts](https://fonts.google.com/specimen/Inter/about))

но если захочешь, можешь определить диапазон, который тебе нужно будет подгрузить:

![image](https://github.com/user-attachments/assets/0f355b9f-440e-4d83-9a15-427de3aecdd1)
подробнее [тут](https://nextjs.org/docs/app/api-reference/components/font#weight)

yaa-ku commented on src/app/globals.css:
```diff
@@ -1,42 +1,277 @@
+  margin: 0;
+  padding: 0;
+  box-sizing: border-box;
+  font-family: "Inter", sans-serif;
+  font-style: normal;
```
соответственно, это нам тут не надо

yaa-ku commented:
важное замечание - если ты используешь картинки (именно картинки, не svg), то используй компонент <Image/> от Next.js

![image](https://github.com/user-attachments/assets/79ae11a0-4aee-44df-a3b8-e7ed94a5fcba)

почитай про работу с картинками [тут](https://nextjs.org/docs/app/building-your-application/optimizing/images#local-images)

SDeeKey committed:
layout + page
(fonts and <main/>)

yaa-ku commented:
господь, выходные все не отпускают :D

yaa-ku commented on src/app/globals.css:
```diff
@@ -1,42 +1,277 @@
-    --foreground: #ededed;
-  }
+/* Стили для заголовков */
+h1, h2, h3, h4, h5, h6 {
+  margin: 0;
+  color: var(--gray-900);
+}
+
+
+
+/*      H1      */
+h1.desktop {
+  font-weight: 700;
+  font-size: 60px; /* 48px */
+  line-height: 72px;
+  letter-spacing: -2px;
```
по стилям: давай уходить от такого. такой подход с привязкой к html-тэгам очень негибкий

семантика - большое искусство. вдруг тебе понадобиться для (с точки зрения html) юзать ссылку, но применить к ней нужно будет стиль кнопки? такое может быть. и лучше ты сам просто пропишешь для этой ссылки класс кнопки, чем страдать от того, что ты уже жестко привязался к типам элементов

также для разных уровней заголовков лучше просто написать отдельные стили, чем переопределять так жестко

yaa-ku commented:
в целом по структуре глобальных стилей и тд предлагаю:
сделать папку стилей, например:
colors.scss <- настраиваешь цветовую палитру. вернее - хранишь тут переменные
drop.scss <- сбрасываешь дефолтные стили браузера
fonts.scss <- можно даже отдельно тут настроить шрифты. но это опционально, просто кидаю варианты
mixins.scss <- твои миксины, чтобы ✨ _жизнь была лучше_  ✨
breakpoints.scss <- брейкпоинты для адаптива в проекте. переменные
global.scss или index.scss <- импортишь туда все вышесказанные файлы. и делаешь ещё нужные тебе штуки

yaa-ku commented:
> в целом по структуре глобальных стилей и тд предлагаю: сделать папку стилей, например: colors.scss <- настраиваешь цветовую палитру. вернее - хранишь тут переменные drop.scss <- сбрасываешь дефолтные стили браузера fonts.scss <- можно даже отдельно тут настроить шрифты. но это опционально, просто кидаю варианты mixins.scss <- твои миксины, чтобы ✨ _жизнь была лучше_ ✨ breakpoints.scss <- брейкпоинты для адаптива в проекте. переменные global.scss или index.scss <- импортишь туда все вышесказанные файлы. и делаешь ещё нужные тебе штуки

а можно ещё отдельно variables всякие. но тут хз, если нужно будет

yaa-ku commented:
пример, как было у меня (не ручаюсь за самый классный в мире пример, это было 1.5 года назад) в глобальном файле стилей

```
body {
  background-color: $background-color;
  color: $primary-text-color;
} <- переопределяю глобальный компонент

.title-1 { <- использую в компонентах с текстом
  font-size: 38px;
  font-weight: bold;

  @media screen and (max-width: 1024px) { <- адаптив. но можно было циферки переменными
    font-size: 30px;
  }

  @media screen and (max-width: 500px) {
    font-size: 28px;
  }
}

.link {
  font-size: 16px;
  font-weight: bold;
  color: $primary-text-color;
  cursor: pointer;

  @media screen and (max-width: 1024px) {
    font-size: 14px;
  }

  &:hover{
    text-decoration: underline;
  }
} <- переопределяю для ссылок
```

yaa-ku commented on src/app/globals.css:
```diff
@@ -1,42 +1,277 @@
+
+
+
+/*      Subtitle      */
+sub.subtitleNormal {
+  font-weight: 400;
+  font-size: 20px;
+  line-height: 28px;
+}
```
не забываем, что sub - не subtitle
это наебка для уебка, а мы не такие

yaa-ku commented on src/app/globals.css:
```diff
@@ -1,42 +1,277 @@
+
+/* md */
+.shadow-md {
+  box-shadow: 0px 2px 2px 0px rgba(0,0,0,7%);
```
по-моему, если 0px можно px не писать. но лучше чекни

SDeeKey replied:
можно не писать, но я это скопировал с сайта с тенями, когда делал их, так что переделаю, это все равно ни на что не влияет

yaa-ku replied:
я потом докопаюсь, когда буду докапываться до форматирования :D

но это на десерт

yaa-ku commented on src/app/globals.css:
```diff
@@ -1,42 +1,277 @@
+/*      H1      */
+h1.desktop {
+  font-weight: 700;
+  font-size: 60px; /* 48px */
```
а че за комментарий? 👀

yaa-ku commented on src/app/globals.css:
```diff
@@ -1,42 +1,277 @@
+label {
+  display: block;
+  font-weight: 500;
+  color: #1f2937;
```
а че за магический цвет?

SDeeKey committed:
+ folder styles
+ global

SDeeKey committed:
+ Button

SDeeKey committed:
+ images

SDeeKey committed:
+ linkIcon

SDeeKey committed:
+ Tag

yaa-ku commented on src/app/globals.scss:
```diff
@@ -0,0 +1,266 @@
+@import '../styles/mixins';
+@import '../styles/breakpoints';
+
+menu-link {
```
а че это без .?
это ж вроде как класс, а не элемент

SDeeKey replied:
Это уже и я увидел)

yaa-ku commented on src/app/globals.scss:
```diff
@@ -0,0 +1,266 @@
+  &:hover {
+    @include heading-styles(900, 16px, 24px);
+    color: $gray-900;
+    text-decoration: none;
```
ты уже поставил такое свойство (text-decoration: none) для .menu-link. нет смысла переопределять его в псевдоклассах, если оно не изменяется

yaa-ku commented on src/app/globals.scss:
```diff
@@ -0,0 +1,266 @@
+  &:active {
+    @include heading-styles(800, 16px, 24px);
+    color: $gray-900;
+    text-decoration: none;
```
тут тоже, излишне

yaa-ku commented on src/app/globals.scss:
```diff
@@ -0,0 +1,266 @@
+  &:hover {
+    @include heading-styles(900, 16px, 24px);
+    color: $gray-900;
+    text-decoration: underline;
+  }
+
+  &:active {
+    @include heading-styles(800, 16px, 24px);
+    color: $gray-900;
+    text-decoration: underline;
```
лишнее свой-во text-decoration

yaa-ku commented on src/styles/_mixins.scss:
```diff
@@ -0,0 +1,36 @@
+}
+
+//Centering mixin
+@mixin center {
+  align-items: center;
+  justify-content: center;
+}
+//Centering start mixin
+@mixin start {
+  align-items: start;
+  justify-content: center;
+}
```
вкусовщина, но вроде не очень осмысленное название. мб лучше align-center или display-center? просто не оч говорящее имхо. что думаешь?

SDeeKey replied:
Если align-center, то это будто выравние только по горизонтали

yaa-ku commented on src/shared/ui-kit/button/button.tsx:
```diff
@@ -1,11 +1,26 @@
 import React from 'react';
-import styles from './button.module.scss';
+import s from './button.module.scss';
+
```
лишний пробел (да, до оформления тоже буду докапываться. но пока не сильно, только то, что бросилось в глаза)

SDeeKey replied:
По оформлению пиши все, что видишь

yaa-ku commented on src/shared/ui-kit/button/button.tsx:
```diff
@@ -1,11 +1,26 @@
+
+const Button = ({title, anchorId}: ButtonProps) => {
+
+    const handleClick = () => {
+        const anchor =  document.getElementById(anchorId);
+        if (anchor) {
+            anchor.scrollIntoView({behavior: 'smooth'});
+        }
+    }
```
это что за зверь у нас? 👀

SDeeKey replied:
Ты о чем?

yaa-ku commented on package.json:
```diff
@@ -9,9 +9,13 @@
     "lint": "next lint"
   },
   "dependencies": {
+    "@svgr/webpack": "^8.1.0",
```
[svgr webpack](https://www.npmjs.com/package/@svgr/webpack) должен был быть установлен как dev-dependencies 

`npm install @svgr/webpack --save-dev`

SDeeKey replied:
Я так и делал

yaa-ku replied:
а почему оно у тебя тогда не в dev-dependencies?

yaa-ku commented on package.json:
```diff
@@ -9,9 +9,13 @@
     "react": "19.0.0-rc-66855b96-20241106",
     "react-dom": "19.0.0-rc-66855b96-20241106",
-    "next": "15.0.3"
+    "webpack": "^5.96.1",
+    "webpack-cli": "^5.1.4"
```
а это здесь откуда? 👀👀👀👀👀 
я чет не оч понимаю, как и зачем ты это поставил. не обязательно пакет вебпака ставить. по крайней мере. мне отдельно ничего устанавливать не пришлоись

SDeeKey replied:
Думаешь я просто так 3 часа просидел?) В среду исправим

yaa-ku replied:
я ещё почувствовала где-то подвох 🌚🌚🌚

yaa-ku commented on package.json:
```diff
@@ -9,9 +9,13 @@
+    "@svgr/webpack": "^8.1.0",
+    "clsx": "^2.1.1",
+    "next": "15.0.3",
     "react": "19.0.0-rc-66855b96-20241106",
     "react-dom": "19.0.0-rc-66855b96-20241106",
```
убери канареек, пж
это новые, кайнда экспериментальные версии. моментами могут быть нестабильными (хотя вроде реакт пишет, что уже стабильные)

но покуда ты не собираешься использовать супер-крутые-новые-классные фичи этих версий, тебе не надо их использовать

ставить сомнительные версии, не фиксировать версии на крупных проектах, апать версии без видимых причин - не очень хорошая практика

SDeeKey replied:
Видимо где-то в ide стоит галочка на пункт “использовать экспериментальные версии”

yaa-ku replied:
yep, возможно. но так как ты по итогу вряд ли собираешься экспериментальные версии использовать - то лучше зафиксируй более стабильную версию

yaa-ku commented on package.json:
```diff
@@ -9,9 +9,13 @@
+    "webpack-cli": "^5.1.4"
   },
   "devDependencies": {
     "typescript": "^5",
```
укажи минорную и патчевую версию

yaa-ku commented on src/app/globals.scss:
```diff
@@ -0,0 +1,266 @@
+.shadow-2xl {
+  box-shadow: 0 25px 25px 0 rgba(0, 0, 0, 0.15);
+}
+
+
+
```
многа пробелов

yaa-ku commented on src/app/globals.scss:
```diff
@@ -0,0 +1,266 @@
+  max-width: 100%;
+}
+
+label {
+  display: block;
+  font-size: 14px;
+  color: $gray-500;
+  margin-bottom: 4px;
+}
+
+input, textarea {
+  font-size: 16px;
+  padding: 12px 16px;
+  width: 252px;
+  height: 48px;
+  outline: none;
+  background: $white;
+  color: $gray-500;
+  border: 1px solid $gray-500;
+  border-radius: 12px;
+  transition: .3s ease;
+  overflow: hidden; // Скрыть переполнение
```
а ето что у нас, почему не классами?

yaa-ku commented on src/app/globals.scss:
```diff
@@ -0,0 +1,266 @@
+}
+
+// Error style
+error {
+  input, textarea {
+    border: 1px solid $red-500;
+  }
+
+  label {
+    color: $red-500;
+  }
+}
```
а че не класс, оно вообще работает?

и опять, обращаешься к элементам

yaa-ku commented on src/shared/ui-kit/button/button.tsx:
```diff
@@ -0,0 +1,29 @@
+
+    return (
+        <div>
+            <button className={`${s['btn']} subtitle-sb`} onClick={handleClick}>
```
фу какая гадость в классах

давай-ка лучше используй clsx. а то если добавишь еще пару классов станет некрасиво и плохо читаемо

yaa-ku commented on src/shared/ui-kit/button/button.tsx:
```diff
@@ -0,0 +1,29 @@
+    }
+
+    return (
+        <div>
```
обертка все ещё лишняя :D

yaa-ku commented on src/shared/ui-kit/linkIcon/linkIcon.tsx:
```diff
@@ -0,0 +1,21 @@
+const LinkIcon = ({link, size = 'md'}: LinkIconProps) => {
+    return (
+        <a href={link} target='_blank' rel='noopener noreferrer' >
+            <div className={clsx(s['button'], s[`button--${size}`])}>
```
а че у нас у linkIcon класс 'button'?

давай ей дадим класс 'link-icon'

SDeeKey replied:
Ну это ж все-таки кнопка) ссылочная кнопка

yaa-ku replied:
но компонент-то называется linkIcon? к тому же, семантично это вообще ссылка у тебя

если настолько хочешь отразить суть того, что это кнопка, можешь обозвать классы для неё как iconButtonLink или че-нибудь такое

а то тут.. вроде компонент linkIcon, под капотом - ссылка, но почему-то стили идут резко для кнопки :D

yaa-ku commented on src/shared/ui-kit/linkIcon/linkIcon.tsx:
```diff
@@ -0,0 +1,21 @@
+const LinkIcon = ({link, size = 'md'}: LinkIconProps) => {
+    return (
+        <a href={link} target='_blank' rel='noopener noreferrer' >
+            <div className={clsx(s['button'], s[`button--${size}`])}>
```
я тут начинаю задумываться, а нам вообще нужен тут div? 
без него прям не получится что ль?

SDeeKey replied:
Как будто все в одну кучу пихать тоже не совсем правильно

yaa-ku replied:
не факт. я потом ещё поковыряюсь над этим компонентом и обсудим. но позже, пока над проектами поработаю

может ты прав, но пока я не совсем вижу, зочем нам это надо

yaa-ku commented on src/shared/ui-kit/tag/tag.module.scss:
```diff
@@ -0,0 +1,15 @@
+  border-radius: 12px;
+  width: fit-content;
+
+  .tag__text {
+    margin: 4px 20px;
+    color: $gray-600;
+  }
```
&__text {
//smth
}

yaa-ku commented on src/shared/ui-kit/tag/tag.tsx:
```diff
@@ -0,0 +1,13 @@
+const Tag = ({title}: {title: string}) => {
+    return (
+        <div className={s['tag']}>
+            <h3 className={`${s['tag__text']} body3-medium`}>{title}</h3>
```
clsx

yaa-ku commented on src/app/globals.scss:
```diff
@@ -0,0 +1,266 @@
+  }
+}
+
+normal-link{
```
я тебе чутка во все места постараюсь потыкать, которые увижу. вот тут тоже

yaa-ku commented on src/app/globals.scss:
```diff
@@ -0,0 +1,266 @@
+  }
+
+}
+
```
лишняя строчка

yaa-ku commented on src/app/globals.scss:
```diff
@@ -0,0 +1,266 @@
+.body3-medium {
+  @include heading-styles(500, 14px, 20px);
+}
+
```
лишняя строчка

yaa-ku commented on src/app/globals.scss:
```diff
@@ -0,0 +1,266 @@
+    color: $red-500;
+  }
+}
+
+
+
```
как много пустых строчек

yaa-ku commented on src/app/page.tsx:
```diff
@@ -1,10 +1,11 @@
-import Image from "next/image";
 import styles from "./page.module.css";
 
+
+
```
пустые строчки

SDeeKey committed:
+input + textarea почистил global

SDeeKey committed:
исправления кнопки, ссылки и тэга

yaa-ku commented on src/app/globals.scss:
```diff
@@ -0,0 +1,204 @@
+@import '../styles/mixins';
+@import '../styles/breakpoints';
+
+.menu-link {
+  @include heading-styles(500, 16px, 24px);
+  color: $gray-600;
+  text-decoration: none;
+
+  &:hover {
+    @include heading-styles(900, 16px, 24px);
+    color: $gray-900;
+  }
+
+  &:active {
+    @include heading-styles(800, 16px, 24px);
+    color: $gray-900;
+  }
+}
```
в данном случае, использование данного миксина не упрощает код

**во-первых**, в рамках псевдоклассов :hover и :active изменяется только параметр веса шрифта, а font-size и line-height остаются прежними, так что излишне переопределять не изменяющиеся параметры. иначе, в случае изменений, например, настроек шрифта, придется менять сразу в нескольких местах

**во-вторых**, в выходном файле css 1 строчка кода вызова миксина превратится в 4 строчки строчки кода. что не очень оптимально

yaa-ku commented on src/app/globals.scss:
```diff
@@ -0,0 +1,204 @@
+  }
+}
+
+.normal-link{
+  @include heading-styles(400, 16px, 24px);
+  color: $gray-600;
+  text-decoration: underline;
+
+  &:hover {
+    @include heading-styles(900, 16px, 24px);
+    color: $gray-900;
+  }
+
+  &:active {
+    @include heading-styles(800, 16px, 24px);
+    color: $gray-900;
+  }
+}
+
```
тк у normal-link и menu-link используют один и те же настройки шрифта, но различаются только весом шрифта, проще вынести общую логику в [placeholder](https://sass-lang.com/documentation/style-rules/placeholder-selectors/)

yaa-ku commented on src/app/globals.scss:
```diff
@@ -0,0 +1,204 @@
+
+// Desktop
+@include respond(desktop) {
+  .heading-h1 {
+    @include heading-styles(700, 60px, 72px, -2%);
+  }
+  .heading-h2 {
+    @include heading-styles(600, 36px, 40px, -2%);
+  }
+  .heading-h3-sb {
+    @include heading-styles(600, 30px, 36px, -2%);
+  }
+  .heading-h3-b {
+    @include heading-styles(700, 30px, 36px, -2%);
+  }
```
1. можно переименовать стиль в heading-h1 например в heading-1. иначе "масло масляное"
2. можно вынести вес шрифтов в переменные. поместить эти переменные, например, можно в файл _variables.scss

```
$normal: 400;
$bold: 700;
// прочее

```

3. letter-spacing: -2% очень часто повторяется, так что его можно также вынести в переменную (кинь в файл _variables.scss)

yaa-ku commented on src/app/globals.scss:
```diff
@@ -0,0 +1,204 @@
+  .heading-h3-b {
+    @include heading-styles(700, 24px, 32px, -2%);
+  }
+  .subtitle-tab-mob-normal {
+    @include heading-styles(400, 18px, 28px);
+  }
+  .subtitle-tab-mob-sb {
+    @include heading-styles(600, 18px, 28px);
+  }
+  .body1-tab-mob {
+    @include heading-styles(400, 16px, 24px);
+  }
```
не стоит выносить в отдельный шрифт настройку адаптива. нужно завести стиль для шрифта и уже в нем рулить параметрами в зависимости от ширины экрана

yaa-ku commented on src/shared/ui-kit/Input/input.module.scss:
```diff
@@ -0,0 +1,57 @@
+
+      &:focus {
+        background: $gray-100;
+        border: 1px solid $orange-500;
```
толщину можно не изменять

yaa-ku commented on src/shared/ui-kit/Input/input.tsx:
```diff
@@ -0,0 +1,91 @@
+        return cleaned; // Если не совпадает, возвращаем очищенный ввод
+    };
+
+    // Писал any вместо React.ChangeEvent<HTMLInputElement> - выдает ошибку, с ошибкой мне не нравится,
```
и правильно, просто так any писать низя. ты правильно сделал!! молодец!!! 👀✨

yaa-ku commented on src/shared/ui-kit/Input/input.tsx:
```diff
@@ -0,0 +1,91 @@
+import s from './input.module.scss';
+import clsx from 'clsx';
+
+type InputType = 'text' | 'password' | 'email' | 'url' | 'phone' | 'login';
```
а url это как?

вообще, если мы делаем такие, универсальные, то это обычно text, password, email, phone. в данном дизайне, конечно, нет, но обычно это ещё и number

yaa-ku commented on src/shared/ui-kit/Input/input.tsx:
```diff
@@ -0,0 +1,91 @@
+
+    // Писал any вместо React.ChangeEvent<HTMLInputElement> - выдает ошибку, с ошибкой мне не нравится,
+    // так что выход такой, если есть идеи чем заменить - напиши
+    const handleChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+        const inputValue = event.target.value;
+        let formattedValue = inputValue;
+
+        // Форматируем значение только для телефонного поля
+        if (type === 'phone') {
+            formattedValue = formatPhoneNumber(inputValue);
+        }
+
+        setValue(formattedValue);
+        const validationMessage = validateInput(formattedValue);
+        setError(typeof validationMessage === 'string' ? validationMessage : '');
```
в целом, обычно по-другому делают. но мне нравится придуманное тобой решение с:
ты хорошо сделал, хорошо продумал. если будут исправления по работе инпутов - потом переделаем в рамках использования новых технологий

yaa-ku commented on src/shared/ui-kit/Input/input.tsx:
```diff
@@ -0,0 +1,91 @@
+            </label>
+            <div className={s['input-wrapper__error-wrapper']}>
+                <input
+                    id="myInput"
```
ой ой ой 👀 
у них получается у всех будет один и тот же id? как-то не хорошечно получается

давай используем для id специальный хук - [useId](https://react.dev/reference/react/useId). так мы гарантируем уникальность сгенерированного id. а ещё это будет считаться более хорошей практикой

SDeeKey replied:
![Screenshot 2024-12-10 154857](https://github.com/user-attachments/assets/6d8421f5-b721-4bf9-b1a3-d7167f50e2eb)
будет ли этот useId гарантировать, что label и input будут синхронизированы?

yaa-ku commented on src/shared/ui-kit/Input/input.tsx:
```diff
@@ -0,0 +1,91 @@
+                    placeholder={type === 'phone' ? '+7(___)-___-__-__' : type}
+                    required
+                />
+                {error && <span className={s['input-wrapper__error']}>{error}</span>}
```
кстати, давай сразу учится разносить более красиво. правка чисто по форматированию

```
{error && 
    <span className={s['input-wrapper__error']}> <- а если несколько параметров, то по разным строчкам лучше
        {error}
    </span>
 }
```

yaa-ku commented on src/shared/ui-kit/button/button.module.scss:
```diff
@@ -0,0 +1,23 @@
+@import "./../../../styles/colors";
+
+.btn {
+  display: flex;
```
у тебя вроде был mixin

yaa-ku commented on src/shared/ui-kit/button/button.module.scss:
```diff
@@ -0,0 +1,23 @@
+@import "./../../../styles/colors";
```
кстати, есть ещё вот такая прикольная штука для упрощения импортов - [path alias](https://blog.logrocket.com/using-path-aliases-cleaner-react-typescript-imports/)
не обязательно сейчас исправлять, уже бы побыстрее пр закрыть, но хотя бы имей ввиду

SDeeKey replied:
знаю, местами использую

yaa-ku commented on src/shared/ui-kit/button/button.tsx:
```diff
@@ -0,0 +1,35 @@
+
+export const Button = ({title, anchorId, onClick}: ButtonProps) => {
+
+    const handleClick = () => {
+        if(anchorId) {
+            const anchor = document.getElementById(anchorId);
+            if (anchor) {
+                anchor.scrollIntoView({behavior: 'smooth'});
+            }
+        }
+        else if (onClick) {
+            onClick();
+        }
+    }
+
+    if (onClick) {
+        onClick(); // Вызов логики из родительского компонента
+    }
```
давай все же не хардкодить поведение кнопки. ui-kit - создание максимально переиспользуемых компонентов, обычно не предполагает закладывание конкретной логики

также для осуществления перелистывания до какого-то элемента лучше использовать не` const anchor = document.getElementById(anchorId)`, а [useRef](https://bobbyhadz.com/blog/react-scroll-to-element-on-click)

SDeeKey replied:
ну а куда надо вставить логику кнопки, если в самой кнопке ее не должно быть?

yaa-ku commented on src/shared/ui-kit/button/button.tsx:
```diff
@@ -0,0 +1,35 @@
+    }
+
+    return (
+        <button className={clsx(s.btn, 'subtitle-sb')} onClick={handleClick}>
```
разнеси на разные строки. красоты и форматирования ради

yaa-ku commented on src/shared/ui-kit/linkIcon/linkIcon.module.scss:
```diff
@@ -0,0 +1,25 @@
+@import "./../../../styles/mixins";
+
+.iconButtonLink {
+  display: flex;
```
вроде был mixin

yaa-ku commented on src/shared/ui-kit/linkIcon/linkIcon.module.scss:
```diff
@@ -0,0 +1,25 @@
+    background-color: $orange-500;
+  }
+  &--md {
+    height:36px;
+    width: 36px;
+
+  }
+  &--lg {
+    height:44px;
+    width: 44px;
+  }
```
щас ещё прикольному свойству научу - [aspect ratio](https://developer.mozilla.org/en-US/docs/Web/CSS/aspect-ratio). а ещё можно КОНКРЕТНО В ЭТОМ ФАЙЛЕ ПРОСТО ДЛЯ ПОВЫШЕНИЯ ЧИТАЕМОСТИ вынести в отдельные переменные

предлагаю:
```
$md-size: 36px;
$lg-size: 44px;

//some awesome code
&--md {
    width: $md-size
    aspect-ratio: 1/1; <- задаем соотношение. получается, задавать высоту уже не нужно - подставится такая же

  }
  &--lg {
    width: $lg-size;
     aspect-ratio: 1/1;
  }
```

yaa-ku commented on src/shared/ui-kit/tag/tag.module.scss:
```diff
@@ -0,0 +1,15 @@
+@import "./../../../styles/colors";
+
+.tag {
+  display: flex;
```
mixin

SDeeKey replied:
на один display: flex у меня нет миксина, у меня он только вместе с flex-direction

yaa-ku commented on src/shared/ui-kit/textarea/textarea.tsx:
```diff
@@ -0,0 +1,49 @@
+'use client'
+
+import React, {useState} from 'react';
```
обычно напрямую испортить реакт необязательно. достаточно 
`import  {useState} from 'react';`

yaa-ku commented on src/shared/ui-kit/textarea/textarea.tsx:
```diff
@@ -0,0 +1,49 @@
+    const validateTextarea = (value: string) => {
+        return value.length <= 500 ? true : 'Character limit';
+    }
+
+
```
строчки 💅

yaa-ku commented on src/shared/ui-kit/textarea/textarea.tsx:
```diff
@@ -0,0 +1,49 @@
+}
+
+export const Textarea = () => {
+    const [value, setValue] = useState<string>('');
+    const [error, setError] = useState<string>('');
```
меня единственное что смущает - значение по умолчанию. может, нам не выставлять его, пусть будет null? тогда мы более корректно будет определять, если у нас нет ошибки или нет значения

но это пока предположение. на подумать

yaa-ku commented on src/styles/_breakpoints.scss:
```diff
@@ -0,0 +1,26 @@
+  @else {
+    @error "Invalid breakpoint name: #{$breakpoint}";
+  }
+}
```
очень хорошая идея, но условия можно чутка переписать. на созвоне говорили

yaa-ku commented on src/styles/_mixins.scss:
```diff
@@ -0,0 +1,36 @@
+  text-overflow: ellipsis;
+}
+// Heading mixin
+@mixin heading-styles($weight, $size, $line-height, $letter-spacing: normal) {
+  font-weight: $weight;
+  font-size: $size;
+  line-height: $line-height;
+  letter-spacing: $letter-spacing;
+}
```
1. не говорящее название у mixin'а
2. сомневаюсь в рациональности и удобности его использования. но круто, что ты с миксинами подружился!

SDeeKey committed:
исправления кнопки, ссылки и тэга

SDeeKey committed:
global

yaa-ku commented on src/shared/ui-kit/Input/input.tsx:
```diff
@@ -0,0 +1,91 @@
+
+interface InputProps {
+    type: InputType;
+    onChange?: (value: string) => void;
```
а где ты это используешь? если не используешь - удали

SDeeKey committed:
last one

SDeeKey committed:
анимацию на кнопке добавил
поменял тег(убрал h3)
лимит поставил на ввод цифр в Phone
дописал * в инпуте и текст арее
доделал текст арею
добавил пробелы в конце файлов

yaa-ku commented on src/app/globals.scss:
```diff
@@ -0,0 +1,220 @@
+}
+
+.menu-link {
+  font-weight: $medium;
+  @extend %link;
```
чисто правка по форматированию: поменяй местами

yaa-ku commented on src/app/globals.scss:
```diff
@@ -0,0 +1,220 @@
+  @include respond(tablet) {
+    font-size: 48px;
+    line-height: 48px;
+  }
```
поставь пустую строчку между

yaa-ku commented on src/app/globals.scss:
```diff
@@ -0,0 +1,220 @@
+    font-size: 36px;
+    line-height: 40px;
+    font-weight: $semibold;
+    letter-spacing: normal;
```
нам это обязательно указывать?

SDeeKey replied:
да, потому что в родителе letter-spacing: $letter-spacing;

yaa-ku commented on src/app/globals.scss:
```diff
@@ -0,0 +1,220 @@
+  font-size: 36px;
+  line-height: 40px;
+  font-weight: $semibold;
+  letter-spacing: $letter-spacing;
```
можно чисто в теории переменную назвать как-нибудь $default-letter-spacing

yaa-ku commented on src/app/globals.scss:
```diff
@@ -0,0 +1,220 @@
+.body-1 {
+  font-size: 18px;
+  line-height: 28px;
+  font-weight: $normal;
```
обзови переменную более говоряще мб. normal-weight мб. но это опционально

yaa-ku commented on src/shared/ui-kit/textarea/textarea.module.scss:
```diff
@@ -0,0 +1,59 @@
+    .textarea-wrapper__textarea {
+      font-size: 16px;
+      padding: 12px 16px;
+      width: 252px;
```
не уверена, что нам нужно так строго задавать ширину. но это уже потом подумаем, пока давай так

yaa-ku commented on src/shared/ui-kit/textarea/textarea.module.scss:
```diff
@@ -0,0 +1,59 @@
+      color: $gray-500;
+      border: 1px solid $gray-500;
+      border-radius: 12px;
+      transition: 0.3s ease;
```
transition можно вынести ай гесс, повторяется не один раз

yaa-ku commented on src/styles/_variables.scss:
```diff
@@ -0,0 +1,8 @@
+// Font weight
+$bold: 700;
+$semibold: 600;
+$medium: 500;
+$normal: 400;
```
к чему доебка выше
с комментарием в файле - понятно что это за переменная. но в целом по проекту переменная normal - очень абстрактное понятие. но не критично

yaa-ku commented:
на данный момент мне все нравится. лучше поправь правки выше, но они не критичные - апрув ставлю и без них (да, ты можешь вносить правки после апрува. можешь например поправить и слить ветку)

ты молодец, все круто! ✨

yaa-ku reviewed (approved):
мир, добро и капибары

SDeeKey committed:
готово
